src/doc/src/reference/pkgid-spec.md

```diff
@@ -38,28 +40,32 @@ that come from different sources such as different registries.
 
 The following are references to the `regex` package on `crates.io`:
 
-| Spec                                                        | Name    | Version |
-|:------------------------------------------------------------|:-------:|:-------:|
-| `regex`                                                     | `regex` | `*`     |
-| `regex@1.4`                                                 | `regex` | `1.4.*` |
-| `regex@1.4.3`                                               | `regex` | `1.4.3` |
-| `https://github.com/rust-lang/crates.io-index#regex`        | `regex` | `*`     |
-| `https://github.com/rust-lang/crates.io-index#regex@1.4.3`  | `regex` | `1.4.3` |
+| Spec                                                              | Name    | Version |
+|:------------------------------------------------------------------|:-------:|:-------:|
+| `regex`                                                           | `regex` | `*`     |
+| `regex@1.4`                                                       | `regex` | `1.4.*` |
+| `regex@1.4.3`                                                     | `regex` | `1.4.3` |
+| `https://github.com/rust-lang/crates.io-index#regex`              | `regex` | `*`     |
+| `https://github.com/rust-lang/crates.io-index#regex@1.4.3`        | `regex` | `1.4.3` |
+| `sparse+https://github.com/rust-lang/crates.io-index#regex@1.4.3` | `regex` | `1.4.3` |
```

Comment:
To be pedantic, this won't work since that isn't a sparse URL (it is a git one).
I don't think there is a way to directly refer to index.crates.io, due to the way it works as a "replaced" source. My only thought is to use something like `sparse+https://example.com/rindex/#regex@1.4.3`, though that would no longer be on "crates.io". It could also just be `sparse+https://index.crates.io/#regex@1.4.3`, though that technically won't work.